manifests/resource.pp

```diff
@@ -19,36 +21,45 @@
 #  [verify_alg] Algorithm used for block validation on peers. Optional.
 #    Defaults to crc32c. Accepts crc32c, sha1, or md5.
 #  [disk_parameters] Parameters for disk{} section
+#  [handlers_parameters] Parameters for handlers{} section
+#  [startup_parameters] Parameters for startup{} section
 #  [manage] If the actual drbd resource shoudl be managed.
 #  [ha_primary] If the resource is being applied on the primary host.
 #  [initial_setup] If this run is associated with the initial setup. Allows a user
 #    to only perform dangerous setup on the initial run.
+#  [initialize] If the actual drbd resource should be initialized
+#  [up] If the actual drbd resource should be set 'up' (drbdadmin up)
 define drbd::resource (
-  $host1           = undef,
-  $host2           = undef,
-  $ip1             = undef,
-  $ip2             = undef,
-  $res1            = undef,
-  $res2            = undef,
-  $cluster         = undef,
-  $secret          = false,
-  $port            = '7789',
-  $device          = '/dev/drbd0',
-  $mountpoint      = "/drbd/${name}",
-  $automount       = true,
-  $owner           = 'root',
-  $group           = 'root',
-  $protocol        = 'C',
-  $verify_alg      = 'crc32c',
-  $rate            = false,
-  $disk_parameters = false,
-  $net_parameters  = false,
-  $manage          = true,
-  $ha_primary      = false,
-  $initial_setup   = false,
-  $fs_type         = 'ext4',
-  $mkfs_opts       = '',
-  $disk            = undef,
+  $host1                                    = undef,
+  $host2                                    = undef,
+  $ip1                                      = undef,
+  $ip2                                      = undef,
+  $res1                                     = undef,
+  $res2                                     = undef,
+  $cluster                                  = undef,
+  $secret                                   = false,
+  $port                                     = '7789',
+  $device                                   = '/dev/drbd0',
+  $mountpoint                               = "/drbd/${name}",
+  $automount                                = true,
+  $owner                                    = 'root',
+  $group                                    = 'root',
+  $protocol                                 = 'C',
+  $verify_alg                               = 'crc32c',
+  $rate                                     = false,
+  $disk_parameters                          = false,
+  $net_parameters                           = false,
+  Optional[Hash] $handlers_parameters       = undef,
```

Comment:
Can you please default to an empty hash and enforce the datatypes for the key/values in it? This is listed as a best practice at https://voxpupuli.org/docs/#reviewing-a-module-pr

Reply:
done, thx for the best practice list.  I just benefit from my own test within this MR: I had to change the condition within the erb template to a check of empty hash instead :)